### PR TITLE
Lesson 7 hw prep

### PR DESCRIPTION
**Pull request by jipariz**
- Students can integrate their already created login screens with prepared network calls as part of their homework 
- The can also make authenticated calls from movieDB so they can fetch real user data from server and finish UserProfileScreen  on which we willl be working on season 7

### CODE REVIEW

File: app/src/main/java/com/strv/movies/MainActivity.kt [hunk 1]
```diff
@@ -85,13 +87,16 @@ class MainActivity : ComponentActivity() {
...
+                                    navController = navController,
+                                    authDataStore = viewModel.authDataStore
+                                )
                             }
                         }
                     ) { paddingValues ->
                         MoviesNavGraph(
                             modifier = Modifier.padding(paddingValues),
-                            navController = navController
+                            navController = navController,
```
Comment (JakubGongol) [inline app/src/main/java/com/strv/movies/MainActivity.kt:99]: New formatter? 😄

File: app/src/main/java/com/strv/movies/database/AuthDataStore.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+private const val AUTH_DATASTORE = "auth_datastore"
+
+
+@Singleton
+class AuthDataStore @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
+    companion object{
+        val SESSION_TOKEN_KEY = stringPreferencesKey(SESSION_TOKEN)
+    }
```
Comment (JakubGongol) [inline app/src/main/java/com/strv/movies/database/AuthDataStore.kt:29]: I would make it **private** and keep it on file level, instead companion object - to be consist with MoviesDataStore

File: app/src/main/java/com/strv/movies/database/AuthDataStore.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+suspend inline fun <T> DataStore<Preferences>.updateValue(key: Preferences.Key<T>, value: T?) {
+    edit {
+        it.update(key, value)
+    }
+}
+
+fun <T> MutablePreferences.update(key: Preferences.Key<T>, value: T?) {
+    value?.let { this[key] = it } ?: remove(key)
+}
```
Comment (JakubGongol) [inline app/src/main/java/com/strv/movies/database/AuthDataStore.kt:68]: This is nice but I'm afraid they'll be confused about why it's done this way now 🤔  - more of a distraction from what we want to learn them 🤷‍♂️  Unless you will go through it with them...

File: app/src/main/java/com/strv/movies/network/NetworkModule.kt [hunk 4]
```diff
@@ -56,9 +70,22 @@ object NetworkModule {
...
+    ): OkHttpClient {
+        return okHttpClient.newBuilder().apply {
+            addInterceptor(authInterceptor)
+        }.build()
+    }
+
     @Provides
     @Singleton
     fun provideLoggingInterceptor(): HttpLoggingInterceptor = HttpLoggingInterceptor { message ->
+        Log.d("HttpLoggingInterceptor", message)
```
Comment (JakubGongol) [inline app/src/main/java/com/strv/movies/network/NetworkModule.kt:88]: 🎖️

File: app/src/main/java/com/strv/movies/network/ApiModule.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+@InstallIn(SingletonComponent::class)
+object ApiModule {
+
+    @Provides
+    @Singleton
+    fun provideMovieApi(@UnauthorizedRetrofit retrofit: Retrofit): MovieApi = retrofit.create()
+
+    @Provides
+    @Singleton
+    fun provideAuthApi(@UnauthorizedRetrofit retrofit: Retrofit): AuthApi = retrofit.create()
```
Comment (JakubGongol) [inline app/src/main/java/com/strv/movies/network/ApiModule.kt:22]: Shouldn't this one be **@AuthorizedRetrofit**?

File: app/src/main/java/com/strv/movies/network/NetworkModule.kt [hunk 6]
```diff
@@ -15,17 +17,28 @@ import retrofit2.Retrofit
...
+import javax.inject.Qualifier
 import javax.inject.Singleton
 
-@Module
-@InstallIn(SingletonComponent::class)
-object ApiModule {
+@Qualifier
+@MustBeDocumented
+@Retention(AnnotationRetention.RUNTIME)
+annotation class UnauthorizedRetrofitOkHttpClient
```
Comment (JakubGongol) [inline app/src/main/java/com/strv/movies/network/NetworkModule.kt:26]: Maybe just: AuthorizedOkHttpClient, UnauthorizedOkHttpClient, AuthorizedRetrofit, UnauthorizedRetrofit 🤔

**Review by JakubGongol — COMMENTED** [review]
I think it's going to be the toughest HW so far :trollface:

File: app/src/main/java/com/strv/movies/network/NetworkModule.kt (hunk 6 above)
Comment (jipariz) [reply app/src/main/java/com/strv/movies/network/NetworkModule.kt:26]: Yeah, that's better

File: app/src/main/java/com/strv/movies/network/ApiModule.kt (hunk 5 above)
Comment (jipariz) [reply app/src/main/java/com/strv/movies/network/ApiModule.kt:22]: Nope. You are not authorized when signing in :D

File: app/src/main/java/com/strv/movies/database/AuthDataStore.kt (hunk 3 above)
Comment (jipariz) [reply app/src/main/java/com/strv/movies/database/AuthDataStore.kt:68]: You are right. I will do it without it

Comment (jipariz) [reply app/src/main/java/com/strv/movies/database/AuthDataStore.kt:68]: You are right. I will do it without it

**Pull request merged**